matrix-sdk/src/main/java/org/matrix/androidsdk/db/MXMediasCache.java

```diff
@@ -294,23 +295,20 @@ public static void clearThumbnailsCache(Context applicationContext) {
     }
 
     /**
-     * Convert matrix url into http one.
+     * Resolve a Matrix media content URI (in the form of "mxc://...") into an HTTP URL.
      *
-     * @param url    the matrix url
+     * @param url    the matrix url, must not be null.
      * @param width  the expected image width
      * @param height the expected image height
-     * @return the URL to access the described resource.
+     * @return the URL to access the described resource, or null if the url is invalid.
      */
+    @Nullable
     private String downloadableUrl(String url, int width, int height) {
```

Comment:
`@NonNull` should be added to the `url` parameter then (can be done later)

Reply:
I will ignore this remark, because the nullity checks are done finally